util/topgen/templates/address_map.md.tpl

```diff
@@ -0,0 +1,73 @@
+<!--
+DO NOT EDIT THIS FILE DIRECTLY.
+It has been generated with the following command:
+${gencmd}
+-->
+
+# ${top["name"].capitalize()} Address Map
+
+% for addr_space in top['addr_spaces']:
+${"##"} ${addr_space['name'].capitalize()} Address Space
+
+% if 'desc' in addr_space:
```

Comment:
I always find these files quite difficult to read ("Give me more curly braces!!"). Sometimes, I've seen things that be slightly easier if they indent the stuff inside. So this line would be `%   if ...` (gaining 2 spaces because it's inside the `for` loop). Definitely not a requirement, but I thought I'd suggest it.